onnxscript/function_libs/torch_aten/ops/common.py

```diff
@@ -0,0 +1,13 @@
+"""Commonly shared functions for the function library."""
+from __future__ import annotations
+
+import onnx.helper
+
+from onnxscript.onnx_opset import opset18 as op
+
+
+def ones_like(x, onnx_dtype: int):
+    shape = op.Shape(x)
+    return op.ConstantOfShape(
+        shape, value=onnx.helper.make_tensor("one", onnx_dtype, [1], [1])
```

Comment:
One issue here is that ONNX attributes should be compile-time (that is, script-time) constants, and, hence, cannot depend on a parameter like onnx_dtype. Is the idea that output type should be the same as input type `x`? Or, can it be a different dtype?

Anyway, I would suggest using either `CastLike` or `Cast`, depending on what exactly you want.

Reply:
The only thing we can do with an attribute-parameter in ONNX is to forward it as the attribute-value of another call.

Reply:
Here's one option:

```py
def ones_like(x, dtype:int):
    shape = op.Shape(x)
    one = op.Constant(value_float=1.0)
    one_dtype = op.Cast(one, to=dtype)
    return op.Expand(one_dtype, shape)
```
I wrote it out in full to focus on the ONNX representation. In onnxscript, we can abbreviate it a little bit, as below:
```py
def ones_like(x, dtype:int):
    shape = op.Shape(x)
    one_dtype = op.Cast(1, to=dtype)  # Forwarding attribute-parameter as attribute-value is ok.
    return op.Expand(one_dtype, shape)
```

Reply:
The constraints come from various ONNX ops.

Reply:
The other point I would like to understand is about the intended usage. In an exporter, for example: you could choose to
* add multiple ONNX ops (a subgraph) to the graph being constructed (representing a single pytorch op)
* add a single call to an ONNX function (op),

They are slightly different (for example, the first approach is more general). The first can be done using an approach we have discussed previously, something like below:
```py
def specialized_ones_like(dtype: int):
   # This is outside the script, so okay:
   one_dtype = onnx.helper.make_tensor("one", dtype, [1], [1])
   @script()
   def f(x):
      shape = op.Shape(x)
      return op.ConstantOfShape(shape, value=one_dtype)
   return f.to_function_proto().node # Depending on intended usage
```

Reply:
Thanks for the example! I love how clean it is. I am probably going to think more about the specialized version before using it requires the onnx function to be defined in a function scope to be able to close over dtype.

Question: `f.to_function_proto().node` looks cool which I didn't know about. What is the `node` we get here?

Reply:
Done